pkg/agent/controller/egress/egress_controller_test.go

```diff
@@ -470,6 +473,18 @@ func TestSyncEgress(t *testing.T) {
 			// Call it one more time to ensure it's idempotent, no extra datapath calls are supposed to be made.
 			err = c.syncEgress(tt.newEgress.Name)
 			assert.NoError(t, err)
+			checkEgressStatus := func(egressName string) {
+				egress, err := c.crdClient.CrdV1alpha2().Egresses().Get(context.TODO(), egressName, metav1.GetOptions{})
+				assert.NoError(t, err)
+				t.Logf("Egress (%s) status: %s", egress.Name, egress.Status.EgressNode)
+				if c.localIPDetector.IsLocalIP(egress.Spec.EgressIP) {
+					assert.Equal(t, c.nodeName, egress.Status.EgressNode, "Egress status not match")
+				} else {
+					assert.Equal(t, "", egress.Status.EgressNode, "Egress status not match")
+				}
```

Comment:
The unit test shouln't calculate the desired status based on the actual status, which basically repeats the main code and not tests it.
You can add `expectedEgresses` and set the expected Node name there, and checks if the got egress equals the expected one. This will be useful later when we have more fields.
```
for _, expectedEgress := range tt.expectedEgresses {
    gotEgress, err := c.crdClient.CrdV1alpha2().Egresses().Get(context.TODO(), expectedEgress.Name, metav1.GetOptions{})
    require.NoError(t, err)
    assert.Equal(t, expectedEgress, gotEgress)
}
```

Reply:
done